File: Java/DesignPatterns/src/main/java/designPatterns/behavioral/visitor/MainVisitor.java
package designPatterns.behavioral.visitor;

public class MainVisitor {
  public static void main(String[] args) {
    Part part = new ConcretePart.A();
    System.out.println(part.accept(new DisplayVisitor()));

    part = new ConcretePart.B();
    System.out.println(part.accept(new DisplayVisitor()));

    part = new ConcretePart.C();
    System.out.println(part.accept(new DisplayVisitor()));
  }
}

File: Kotlin/DesignPatterns/src/main/kotlin/behavioral/visitor/MainVisitor.kt
package behavioral.visitor

fun main() {
  var part: Part = ConcretePart.A()
  println(part.accept(DisplayVisitor()))

  part = ConcretePart.B()
  println(part.accept(DisplayVisitor()))

  part = ConcretePart.C()
  println(part.accept(DisplayVisitor()))
}